sigridci/sigridci.py

```diff
@@ -140,17 +141,19 @@ def retry(self, operation, *, attempts=5, allow404=False, allowEmpty=True):
 
     def submitUpload(self, options, systemExists):
         log("Creating upload")
-        uploadPacker = SystemUploadPacker(options)
-        upload = "sigrid-upload-" + datetime.datetime.now().strftime("%Y%m%d-%H%M%S") + ".zip"
-        uploadPacker.prepareUpload(options.sourceDir, upload)
-
-        log("Preparing upload")
-        uploadLocation = self.obtainUploadLocation(systemExists)
-        uploadUrl = uploadLocation["uploadUrl"]
-        analysisId = uploadLocation["ciRunId"]
-        log(f"Sigrid CI analysis ID: {analysisId}")
-        log("Publishing upload" if self.publish else "Submitting upload")
-        self.uploadBinaryFile(uploadUrl, upload)
+        
+        with tempfile.NamedTemporaryFile() as tempFile:
+            uploadPacker = SystemUploadPacker(options)        
+            uploadPacker.prepareUpload(options.sourceDir, tempFile.name)
```

Comment:
I believe neither `tempfile.NamedTemporaryFile()` nor making a zip-archive out of the file in `uploadPacker.prepareUpload` adds the `.zip` extension to the file. So that makes the test checking for any files ending with `.zip` not very useful.

Although if I'm not mistaken the name of this (temporary) file is not important for the actual upload via the request, it wouldn't hurt either to have it end in `.zip` and that would clearly show the intention. The test would still be mainly testing the behaviour of `tempfile`, but as it still documents the desired behaviour/conscious decision I think that's fine. I believe the rest of the name we gave the original persisting `upload` file is not that important.

So I would suggest to add a suffix.

```suggestion
        with tempfile.NamedTemporaryFile(suffix=".zip") as tempFile:
            uploadPacker = SystemUploadPacker(options)        
            uploadPacker.prepareUpload(options.sourceDir, tempFile.name)
```